working folder for group collab

xibz opened the pull request:
Session #1 

@sbrennan4

sbrennan4 commented on ticket-proposal/tickets_idea_and_questions.md:
```diff
@@ -0,0 +1,201 @@
+
+* Created/Open
+* InProgress (from the development point of view)
+* Reviewed
```
```suggestion
* Peer Review
```

sbrennan4 commented on ticket-proposal/tickets_idea_and_questions.md:
```diff
@@ -0,0 +1,201 @@
+    - approval.declined
+    - approval.inprogress
+    - approval.approved
+* Testing
```
```suggestion
* QA Testing
```

xibz replied:
Why do we need to make a distinction between `Testing` and `QA Testing`?

sbrennan4 commented on ticket-proposal/tickets_idea_and_questions.md:
```diff
@@ -0,0 +1,201 @@
+* Created/Open
+* InProgress (from the development point of view)
+* Reviewed
+* Approval statuses
+    - approval.declined
+    - approval.inprogress
+    - approval.approved
```
Maybe we don't need these sub-states.

```suggestion
* Approval/Sign-off
* Ready to Deploy
```

`Approval` meaning it is waiting for approval. If approved, it goes to either `Ready to Deploy` or `Testing` or some other state depending on the team's workflow. If rejected, it goes back to `Open` or `In progress` or some other state.

sbrennan4 commented on ticket-proposal/tickets_idea_and_questions.md:
```diff
@@ -0,0 +1,201 @@
+    - approval.inprogress
+    - approval.approved
+* Testing
+* Resolved
```
Should we include a state for `Deployment`? Could be required depending on the definition of done.

```suggestion
* Deployment
* Resolved
```

sbrennan4 replied:
```suggestion
* Resolved/Done
```

sbrennan4 reviewed (commented):
Some thoughts on ticket states. I think it makes the most sense to have events when a ticket transitions _to_ a state and not necessarily when a ticket transitions _from_ a state so we should update our naming based on that perspective and use the present tense.

xibz commented:
> Thinking about this further. The CDEvents vocabulary uses past tense for the predicates so if we're trying to map these states to predicates then we should keep that in mind.

CDEvents are declarative, so it's whatever occurred from some service. However,. that doesn't we can't use present tense here, since what did occur was moving to that state. So I think changing them to be present tense makes sense

xibz commented:
@sbrennan4 from SIG meeting CDEvents 10/23/2023

Ticket should have predicates: created, updated, and resolved
Updated with state free form

Have stringed enums for the `updated` field, but also allow it to be freeform:
```go
type TicketState string

const (
    Created TicketState = "created"
    InProgress TicketState = "created"
    Closed TicketState = "closed"
)

ticketEvent := ticket.NewTicket(ticket.Created) // ticket.NewTicket("something-custom")
```

```cdevents
{
    "subject": {
        "state": "created" or "something-custom"
    }
}
```

Approval may have score to indicate whether or not a manual approval is needed

xibz commented:
Link to ticket proposal https://hackmd.io/@Mev468KqSlSu2GZpMzXKhQ/ByYuaEufa

sbrennan4 commented:
TODO: go over jira api and make sure everything fits into the spec
